tests/test_kb_dataset.py

```diff
@@ -0,0 +1,310 @@
+import argparse
+import sys
+import unittest
+from collections import defaultdict
+from difflib import ndiff
+from pathlib import Path
+from pprint import pformat
+from typing import Iterator, Optional, Union
+
+sys.path.append(str(Path(__file__).parent.parent))
+
+import datasets
+
+from schemas.kb import features
+
+OFFSET_ERROR_MSG = (
+    "\n"
+    "There are features with wrong offsets!"
+    "This is not a hard failure, as it is common for this type of datasets"
+    "However, if the error list is long (e.g. >10) you should double check your code"
+)
+
+
+def _get_example_text(example: dict) -> str:
+    """
+    Get full text example
+    """
+    return " ".join([t for p in example["passages"] for t in p["text"]])
```

Comment:
I think I prefer `"".join` as opposed to `" ".join` ... this way passages is a partition of the text with no gaps and the offsets for the passages wont have any gaps ... we could also add a test for that .. something like, 
```
example['passages'][n]['offsets'][0][1] = example['passages'][n-1]['offsets'][0][0] + 1
```
what do you think?

Reply:
I used the whitespace to join passages because from my experience this is sort of a convention of datasets in PubTator and BioC format. We can make "passages" contiguous but I think we need to say it explicitly somewhere then, as it might be in the "muscle memory" of people like me. 

Thumbs up on the test anyway. I will add as soon as we agree on this.

Reply:
sounds good. if its more of a convention in bio datasets let's keep as `" "`